client/src/js/services/SearchModal.js

```diff
@@ -0,0 +1,33 @@
+angular.module('bhima.services')
+  .service('SearchModalUtilService', SearchModalUtilService);
+
+SearchModalUtilService.$inject = [];
+
+function SearchModalUtilService() {
+  const service = this;
+
+  service.submit = (instance, searchQueries, changes, displayValues, lastDisplayValues) => {
```

Comment:
It looks like `service.submit()` is the same as `service.applyChanges()`, but it also closes the modal.

It also looks like you only wrote these two functions to account for one modal in the [DisplayMetadataSearchModalController](https://github.com/IMA-WorldHealth/bhima/pull/4691/files#diff-2afefe47ae8f9b1a1dc2e769bb8a5c2bL83).

I propose you make this one function `getChanges()` that returns `changes.getAll()`.  That way you don't have to pass in `instance` into the function, and the function isn't responsible for closing the modal.